### PR TITLE
DM-12384: exit/raise when data is less than parameters

### PR DESCRIPTION
**Pull request by parejkoj**

### CODE REVIEW

File: tests/test_astrometryModel.py [hunk 1]
```diff
@@ -390,6 +390,12 @@ def testGetVisitTransform(self):
         self.checkGetVisitTransform(self.model1)
         self.checkGetVisitTransform(self.model2)
 
+    def test_validate(self):
+        # Model1 has 70 parameters (2 visits*20 params + (6-1) sensors*5 params)
+        # so we need 70+1 degrees of freedom to fit it.
```
Comment (mrawls) [inline tests/test_astrometryModel.py]: This isn't quite the right use of "degrees of freedom," I don't think; you need at least 1 degree of freedom so you need 71 data points/measurements/observations/inputs.

File: tests/test_photometryModel.py [hunk 2]
```diff
@@ -322,10 +322,12 @@ def test_checkPositiveOnBBox(self):
         self.assertFalse(self.model.checkPositiveOnBBox(self.ccdImageList[0]))
 
     def test_validate(self):
-        self.assertTrue(self.model.validate(self.ccdImageList))
-        # Make the model go negative
+        # Model has 20 parameters (2 visits, 10 params each)
```
Comment (mrawls) [inline tests/test_photometryModel.py]: Please add a comment like in the astrometryModel test that explains why the assertion with 1 dof should be True but the one with 0 dof should be False.

File: tests/test_photometryModel.py [hunk 3]
```diff
@@ -322,10 +322,12 @@ def test_checkPositiveOnBBox(self):
...
+        self.assertTrue(self.model.validate(self.ccdImageList, 1))
+        self.assertFalse(self.model.validate(self.ccdImageList, 0))
+        # Make the model go negative on the bounding box
         self.model.offsetParams(-5*self.delta)
-        self.assertFalse(self.model.validate(self.ccdImageList))
+        self.assertFalse(self.model.validate(self.ccdImageList, 100))
```
Comment (mrawls) [inline tests/test_photometryModel.py]: Can you explain what situation this line is testing? What does the offset do, and why have you put 100 dof?

File: python/lsst/jointcal/jointcal.py [hunk 4]
```diff
@@ -737,7 +737,7 @@ def _logChi2AndValidate(self, associations, fit, model, chi2Label="Model",
         self._check_stars(associations)
         if not np.isfinite(chi2.chi2):
             raise FloatingPointError('%s chi2 is invalid: %s', chi2Label, chi2)
-        if not model.validate(associations.getCcdImageList()):
+        if not model.validate(associations.getCcdImageList(), chi2.ndof):
             raise ValueError("Model is not valid: check log messages for warnings.")
```
[the remark this answers is not in the document]
Comment (parejkoj) [reply python/lsst/jointcal/jointcal.py]: As it stands, this code doesn't know anything about why the model did not validate, just that it didn't. The call to `validate` should produce `ERROR` messages in the log that will be more descriptive.

File: tests/test_astrometryModel.py (hunk 1 above)
Comment (parejkoj) [reply tests/test_astrometryModel.py]: Yeah, that was left over from when I was still sorting out what "degrees of freedom" meant in this context. I'm a bit disappointed that `validate` is just checking that the number is one or more, but I guess that's all there is to check. I'll reword it.

Relatedly, can we chat about whether that number should be 1 or 2 for astrometry?

File: tests/test_astrometryModel.py [hunk 5]
```diff
@@ -390,6 +390,12 @@ def testGetVisitTransform(self):
         self.checkGetVisitTransform(self.model1)
         self.checkGetVisitTransform(self.model2)
 
+    def test_validate(self):
```
[the remark this answers is not in the document]
Comment (parejkoj) [reply tests/test_astrometryModel.py]: It's testing the `validate` method. I've added a docstring and some comments.

**Comment by parejkoj** [comment]
Does the update help with clarity?

**Pull request merged**